src/app/store/machine/types/actions.ts

```diff
@@ -254,54 +254,53 @@ export enum FetchSortDirection {
 type ArrayOrValue<T> = { [P in keyof T]: T[P] | Array<T[P]> };
 
 type Filters = ArrayOrValue<{
-  free_text: string;
-  system_id: Machine["system_id"];
-  hostname: Machine["hostname"];
+  agent_name: string;
+  arch: Machine["architecture"];
+  cpu_count: Machine["cpu_count"];
+  cpu_speed: Machine["cpu_speed"];
   description: Machine["description"];
   distro_series: Machine["distro_series"];
-  osystem: Machine["osystem"];
-  error_description: Machine["error_description"];
-  mac_address: NetworkInterface["mac_address"];
   domain: Domain["name"];
-  agent_name: string;
+  error_description: Machine["error_description"];
   fabric_classes: Fabric["class_type"];
   fabrics: Machine["fabrics"];
-  zone: Machine["zone"]["name"];
+  free_text: string;
+  hostname: Machine["hostname"];
+  ip_addresses: NodeIpAddress["ip"];
+  link_speed: Machine["link_speeds"][0];
+  mac_address: NetworkInterface["mac_address"];
+  mem: Machine["memory"];
+  osystem: Machine["osystem"];
+  owner: Machine["owner"];
+  pod_type: Pod["type"];
+  pod: ModelRef["name"];
   pool: ResourcePool["name"];
-  arch: Machine["architecture"];
+  spaces: Space["name"];
+  status: FetchNodeStatus;
+  subnets: Subnet["name"];
+  system_id: Machine["system_id"];
   tags: Tag["name"];
   vlans: NodeVlan["name"];
-  pod: ModelRef["name"];
-  pod_type: Pod["type"];
-  owner: Machine["owner"];
-  subnets: Subnet["name"];
-  ip_addresses: NodeIpAddress["ip"];
-  spaces: Space["name"];
   workloads: string;
-  status: FetchNodeStatus;
-}> & {
-  mem: string;
-  cpu_count: string;
-  cpu_speed: string;
-  link_speed: string;
-};
```

Comment:
This was mostly ordering by alphabet, but the `mem`, `cpu_count` etc. filters have been moved into the `ArrayOrValue` block as the API now supports arrays for all the filters.